substrate/frame/support/src/generate_genesis_config.rs

```diff
@@ -208,89 +217,154 @@ pub fn retain_initialized_fields(
-				$all_keys.push(
-					$crate::generate_genesis_config::InitializedField::full(concat!(stringify!($key), ".", stringify!($keyi)))
-				);
-				$($type)::+ {
-					$keyi:$value,
-					..Default::default()
-				}
+	($($struct_type:ident)::+, $all_keys:ident @ $key:ident:  $($type:ident)::+ { $($body:tt)* } ) => {
+		(
+			$($struct_type)::+ {
+				$key: {
+					let mut __inner_keys =
+						$crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
+					let __value = $crate::build_struct_json_patch!($($type)::+, __inner_keys @ { $($body)* });
+					for i in __inner_keys.iter_mut() {
+						i.add_prefix(stringify!($key));
+					};
+					$all_keys.push((__value.1,stringify!($key)).into());
+					$all_keys.extend(__inner_keys);
+					__value.0
+				},
+				..Default::default()
 			},
-			..Default::default()
+			$crate::generate_genesis_config::InitilizationType::Partial
+		)
+	};
+	($($struct_type:ident)::+, $all_keys:ident @ $key:ident:  $($type:ident)::+ { $($body:tt)* },  $($tail:tt)*) => {
+		{
+			let mut __initialization_type;
+			(
+				$($struct_type)::+ {
+					$key : {
+						let mut __inner_keys =
+							$crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
+						let __value = $crate::build_struct_json_patch!($($type)::+, __inner_keys @ { $($body)* });
+						$all_keys.push((__value.1,stringify!($key)).into());
+
+						for i in __inner_keys.iter_mut() {
+							i.add_prefix(stringify!($key));
+						};
+						$all_keys.extend(__inner_keys);
+						__value.0
+					},
+					.. {
+						let (__value, __tmp) =
+							$crate::build_struct_json_patch!($($struct_type)::+, $all_keys @ $($tail)*);
+						__initialization_type = __tmp;
+						__value
+					}
+				},
+				__initialization_type
+			)
 		}
 	};
-	($($struct_type:ident)::+, $all_keys:ident  @  $key:ident:  $($type:ident)::+ { $($body:tt)* } ) => {
-		$($struct_type)::+ {
-			$key: {
-				$all_keys.push($crate::generate_genesis_config::InitializedField::partial(stringify!($key)));
-				let mut inner_keys = $crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
-				let value = $crate::build_struct_json_patch!($($type)::+, inner_keys @ { $($body)* });
-				for i in inner_keys.iter_mut() {
-					i.add_prefix(stringify!($key));
-				};
-				$all_keys.extend(inner_keys);
-				value
-			},
-			..Default::default()
+	($($struct_type:ident)::+, $all_keys:ident @ $key:ident: $value:expr, $($tail:tt)* ) => {
+		{
+			let mut __initialization_type;
+			(
+				$($struct_type)::+ {
+					$key: {
+						$all_keys.push($crate::generate_genesis_config::InitializedField::full(
+							stringify!($key))
+						);
+						$value
+					},
+					.. {
+						let (__value, __tmp) =
+							$crate::build_struct_json_patch!($($struct_type)::+, $all_keys @ $($tail)*);
+						__initialization_type = __tmp;
+						__value
+					}
+				},
+				__initialization_type
```

Comment:
This is because the order of execution can be important isn't it? then we should also do it at the line 323

Reply:
ah yes indeed, all good